packages/roosterjs-content-model/lib/publicApi/utils/selectWord.ts

```diff
@@ -0,0 +1,103 @@
+import { ContentModelBlockGroup } from '../../publicTypes/group/ContentModelBlockGroup';
+import { ContentModelSegment } from '../../publicTypes/segment/ContentModelSegment';
+import { ContentModelText } from '../../publicTypes/segment/ContentModelText';
+import { createText } from '../../modelApi/creators/createText';
+
+export function selectWord(
+    path: ContentModelBlockGroup[],
+    marker: ContentModelSegment
+): ContentModelSegment[] {
+    const parent = path[0];
+    const block = parent.blocks[0];
+
+    if (block.blockType == 'Paragraph') {
+        const segments: ContentModelSegment[] = [];
+        let markerSelectionIndex = block.segments.indexOf(marker);
+        for (let i = markerSelectionIndex - 1; i >= 0; i--) {
+            const currentSegment = block.segments[i];
+            if (currentSegment.segmentType == 'Text') {
+                const found = findDelimiter(currentSegment, false);
+                if (found != null) {
+                    if (found == currentSegment.text.length) {
+                        break;
+                    }
+                    splitTextSegment(block.segments as ContentModelText[], i, found);
+                    //block.segments[i + 1].isSelected = true;
+                    segments.push(block.segments[i + 1]);
+                    break;
+                } else {
+                    segments.push(block.segments[i]);
+                }
+            } else {
+                break;
+            }
+        }
+        markerSelectionIndex = block.segments.indexOf(marker);
+        for (let i = markerSelectionIndex + 1; i < block.segments.length; i++) {
+            const currentSegment = block.segments[i];
+            if (currentSegment.segmentType == 'Text') {
+                const found = findDelimiter(currentSegment, true);
+                if (found != null) {
+                    if (found == 0) {
+                        break;
+                    }
+                    splitTextSegment(block.segments as ContentModelText[], i, found);
+                    //block.segments[i].isSelected = true;
+                    segments.push(block.segments[i]);
+                    break;
+                } else {
+                    segments.push(block.segments[i]);
+                }
+            } else {
+                break;
+            }
+        }
+        return segments;
+    } else {
+        return [];
+    }
+}
+
+const PUNCTUATION_REGEX = /[.,:!?()\[\]\\/]/gu;
+const SPACES_REGEX = /[\u00A0\u1680​\u180e\u2000\u2009\u200a​\u200b​\u202f\u205f​\u3000\s\t\r\n]/gm;
+
+function findDelimiter(segment: ContentModelText, moveRightward: boolean): number | null {
+    const word = segment.text;
+    let offset = null;
```

Comment:
You can return -1 as "not found" to follow the same pattern with other array searching functions